### PR TITLE
test: Add comprehensive tests for `OllamaChatModel` and `OllamaEmbeddingModel`

### PR DESCRIPTION
**Pull request by alxkm**
## Summary
This PR adds extensive test coverage for OllamaChatModel and OllamaEmbeddingModel to improve robustness and catch edge cases.

### OllamaChatModel Tests
- **Builder validation**: Tests for null API validation and complete builder options
- **Metadata handling**: Edge cases with zero, large, and null values
- **Aggregation scenarios**: Complex metadata aggregation and overflow handling  
- **Model configuration**: Parameterized tests for different OllamaModel types
- **Data preservation**: Model name and timestamp handling in metadata
- **Immutability**: Verification that builder creates distinct instances

### OllamaEmbedding Tests  
- **Input handling**: Empty inputs, multiple inputs, and input order preservation
- **Options override**: Runtime options properly overriding defaults
- **Keep-alive formats**: Different duration formats (seconds, hours, etc.)
- **Minimal configuration**: Tests with bare minimum settings
- **Whitespace handling**: Preservation of whitespace-only and empty inputs

## Test Coverage Improvements
- Edge case handling for boundary values and null inputs
- Robust validation of configuration options and their precedence
- Data integrity verification for metadata aggregation
- Comprehensive input format testing

## Quality Assurance
- All tests follow existing patterns using AssertJ assertions
- Maintains consistency with current test structure
- Uses appropriate mocking with Mockito where needed
- Includes parameterized tests for efficiency

These tests help ensure the Ollama integration handles various real-world scenarios correctly and maintains backward compatibility.

### CODE REVIEW

File: models/spring-ai-ollama/src/test/java/org/springframework/ai/ollama/OllamaChatModelTests.java [hunk 1]
```diff
@@ -38,6 +40,7 @@
 import org.springframework.ai.retry.RetryUtils;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
 import static org.junit.jupiter.api.Assertions.*;
```
Comment (sobychacko) [inline models/spring-ai-ollama/src/test/java/org/springframework/ai/ollama/OllamaChatModelTests.java]: No wildcard imports.

Comment (alxkm) [reply models/spring-ai-ollama/src/test/java/org/springframework/ai/ollama/OllamaChatModelTests.java]: Looks like it was before. Fixed.

**Review by sobychacko — APPROVED** [review]
